test/e2e_node/volume_manager_test.go

```diff
@@ -0,0 +1,125 @@
+	"k8s.io/kubernetes/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = framework.KubeDescribe("Kubelet Volume Manager", func() {
+	f := framework.NewDefaultFramework("kubelet-volume-manager")
+	Describe("Volume Manager", func() {
+		Context("On terminatation of pod with memory backed volume", func() {
+			It("should remove the volume from the node", func() {
+				var (
+					memoryBackedPod *api.Pod
+					volumeName      string
+				)
+				By("Creating a pod with a memory backed volume that exits success without restart", func() {
+					volumeName = "memory-volume"
+					memoryBackedPod = f.PodClient().Create(&api.Pod{
+						ObjectMeta: api.ObjectMeta{
+							Name:      "pod" + string(uuid.NewUUID()),
+							Namespace: f.Namespace.Name,
+						},
+						Spec: api.PodSpec{
+							RestartPolicy: api.RestartPolicyNever,
+							Containers: []api.Container{
+								{
+									Image:   "gcr.io/google_containers/busybox:1.24",
+									Name:    "container" + string(uuid.NewUUID()),
+									Command: []string{"sh", "-c", "echo"},
+									VolumeMounts: []api.VolumeMount{
+										{
+											Name:      volumeName,
+											MountPath: "/tmp",
+										},
+									},
+								},
+							},
+							Volumes: []api.Volume{
+								{
+									Name: volumeName,
+									VolumeSource: api.VolumeSource{
+										EmptyDir: &api.EmptyDirVolumeSource{Medium: api.StorageMediumMemory},
+									},
+								},
+							},
+						},
+					})
+					err := framework.WaitForPodSuccessInNamespace(f.ClientSet, memoryBackedPod.Name, f.Namespace.Name)
+					Expect(err).NotTo(HaveOccurred())
+				})
+				By("Verifying the memory backed volume was removed from node", func() {
+					volumePath := fmt.Sprintf("/tmp/%s/volumes/kubernetes.io~empty-dir/%s", string(memoryBackedPod.UID), volumeName)
+					var err error
+					for i := 0; i < 10; i++ {
+						// need to create a new verification pod on each pass since updates
+						//to the HostPath volume aren't propogated to the pod
+						pod := f.PodClient().Create(&api.Pod{
+							ObjectMeta: api.ObjectMeta{
+								Name:      "pod" + string(uuid.NewUUID()),
+								Namespace: f.Namespace.Name,
+							},
+							Spec: api.PodSpec{
+								RestartPolicy: api.RestartPolicyNever,
+								Containers: []api.Container{
+									{
+										Image:   "gcr.io/google_containers/busybox:1.24",
+										Name:    "container" + string(uuid.NewUUID()),
+										Command: []string{"sh", "-c", "if [ -d " + volumePath + " ]; then exit 1; fi;"},
+										VolumeMounts: []api.VolumeMount{
+											{
+												Name:      "kubelet-pods",
+												MountPath: "/tmp",
+											},
+										},
+									},
+								},
+								Volumes: []api.Volume{
+									{
+										Name: "kubelet-pods",
+										VolumeSource: api.VolumeSource{
+											HostPath: &api.HostPathVolumeSource{Path: "/var/lib/kubelet/pods"},
```

Comment:
Can you put a // TODO to pull this value from the test context for node (it needs to be added)